data_structures/sparse_table.cpp

```diff
@@ -0,0 +1,141 @@
+#include <array>     /// for std::array
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+/**
+ *
+ * @file
+ * @brief Implementation of Sparse Table
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂